compiler/noirc_evaluator/src/brillig/mod.rs

```diff
@@ -89,17 +92,25 @@ impl Ssa {
             return brillig;
         }
 
+        let used_globals_map = std::mem::take(&mut self.used_globals);
+        let mut brillig_globals =
+            BrilligGlobals::new(&self.functions, used_globals_map, self.main_id);
+
         // Globals are computed once at compile time and shared across all functions,
         // thus we can just fetch globals from the main function.
         let globals = (*self.functions[&self.main_id].dfg.globals).clone();
-        let (artifact, brillig_globals, globals_size) =
-            convert_ssa_globals(enable_debug_trace, globals, &self.used_global_values);
-        brillig.globals = artifact;
-        brillig.globals_memory_size = globals_size;
+        let globals_dfg = DataFlowGraph::from(globals);
+        brillig_globals.declare_globals(&globals_dfg, &mut brillig, enable_debug_trace);
 
         for brillig_function_id in brillig_reachable_function_ids {
+            let mut globals_allocations = HashMap::default();
+            let entry_point_allocations = brillig_globals.get_brillig_globals(brillig_function_id);
+            for map in entry_point_allocations {
+                globals_allocations.extend(map);
+            }
```

Comment:
Should `get_brillig_global` return a `HashMap` directly? This is the only place we call it afaik.

Reply:
yup that is cleaner. `get_brillig_global` returns a map now.